RL/Coordinate system transforms

ralatsdc opened the pull request.

Commit:
Blacken buffer

Commit:
Add tested coordinate system transforms based on quaternions

robotastic commented:
IT IS HAPPENING!

mchadwick-iqt commented:
Hi @ralatsdc - thanks for the PR.  I'm in progress with reviewing, but a couple questions - 

1. We've used pip to install within our Docker environment, it looks like you dropped in a Conda requirements file.  Is there a key reason to switch to Conda?  If no, please convert this to be a pip install via requirements.txt
2. Does this PR replace the original functionality or just put the hooks in for the new functionality?  That'll change how I do hardware in the loop testing
3. Has this been tested within a container yet or is that still to be accomplished?  We can test it here as well just wanted to check
4. It looks like some of the code was converted to be in feet instead of meters.  Was this a bug in the system that you found?

ralatsdc commented:
> Hi @ralatsdc - thanks for the PR. I'm in progress with reviewing, but a couple questions -
@mchadwick-iqt , answers in line ...
> 
> 1. We've used pip to install within our Docker environment, it looks like you dropped in a Conda requirements file.  Is there a key reason to switch to Conda?  If no, please convert this to be a pip install via requirements.txt

No important reason. Switched to using only `requirements.txt`.

> 2. Does this PR replace the original functionality or just put the hooks in for the new functionality?  That'll change how I do hardware in the loop testing

I have hardcoded use of the new function `calculateCameraPositionB()` instead of the old function `calculateCameraPosition()`, which has now been renamed to `calculateCameraPositionA()`. The old function will still work, however, invoking it will require a code modification.

> 3. Has this been tested within a container yet or is that still to be accomplished?  We can test it here as well just wanted to check.

Yes. Added test data to `axis-ptz/data`, and modified `axis-ptz/dockerfile` to copy test data into `/app/data/`.

> 4. It looks like some of the code was converted to be in feet instead of meters.  Was this a bug in the system that you found?

No. The test data does not use the specified units of measure. I have converted all calculations to use the specified units of measure, in particular, meters and degrees. Conversion only happens when using the test data.
